runtime-light/stdlib/crypto/crypto-functions.cpp

```diff
@@ -115,3 +110,175 @@ task_t<int64_t> f$openssl_verify(const string &data, const string &signature, co
 
   co_return 1;
 }
+
+Optional<tl::CipherAlgorithm> parse_algorithm(const string &method) {
+  if (strncasecmp(method.c_str(), "aes-128-cbc", strlen("aes-128-cbc")) == 0) {
```

Comment:
Let's convert it to something more high-level:
```
if (std::ranges::equal(lhs, rhs, ichar_equals)) {...}
```
where 'lhs' and 'rhs' are `std::string_view`s